_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_interlace.cpp

```diff
@@ -234,6 +234,17 @@ void Interlace::Query1NoCaps(const FeatureBlocks& , TPushQ1 Push)
             return prev(par, task, bRAPIntra && !IsField(par.mvp.mfx.FrameInfo.PicStruct));
         });
 
+        defaults.GetVPS.Push([](
+            Defaults::TGetVPS::TExt prev
+            , const Defaults::Param& par
+            , VPS& vps)
+        {
+            vps.general.progressive_source_flag    = !!(par.mvp.mfx.FrameInfo.PicStruct & MFX_PICSTRUCT_PROGRESSIVE);
```

Comment:
Interlace feature supposed to implement only delta to GetVPS, so need to call prev/base code first:
auto sts = prev(par, vps);
...

Reply:
Got it